ui/src/reducers/util.js

```diff
@@ -31,7 +31,10 @@ export function loadComplete(data) {
 }
 
 export function objectLoadComplete(state, id, data = {}) {
-  return { ...state, [id]: loadComplete(data) };
+  const oldData = state[id] || {};
+  const newData = loadComplete(data);
+
+  return { ...state, [id]: { ...oldData, ...newData } };
```

Comment:
It's difficult to see, but I would assume there is no chance of duplication or overwriting here?

Reply:
To answer my own question... The spread operator will, in the case of duplicate items in an object, take the value from the 2nd object.